newIDE/app/src/MarketingPlans/MarketingPlans.js

```diff
@@ -93,55 +93,43 @@ const MarketingPlans = ({ game }: Props) => {
     null
   );
 
-  const {
-    activeBasicFeaturing,
-    activeProFeaturing,
-    activePremiumFeaturing,
-  }: {|
-    activeBasicFeaturing: ?GameFeaturing,
-    activeProFeaturing: ?GameFeaturing,
-    activePremiumFeaturing: ?GameFeaturing,
-  |} = React.useMemo(
+  const activeGameFeaturings: ?(GameFeaturing[]) = React.useMemo(
     () => {
-      if (!gameFeaturings)
-        return {
-          activeBasicFeaturing: null,
-          activeProFeaturing: null,
-          activePremiumFeaturing: null,
-        };
-
-      const activeGameFeaturings = gameFeaturings.filter(
+      if (!gameFeaturings) return null;
+
+      return gameFeaturings.filter(
         gameFeaturing => gameFeaturing.expiresAt > Date.now() / 1000
       );
-      return {
-        activeBasicFeaturing: activeGameFeaturings.filter(
-          gameFeaturing => gameFeaturing.featuring === 'games-platform-home'
-        )[0],
-        activeProFeaturing: activeGameFeaturings.filter(
-          gameFeaturing => gameFeaturing.featuring === 'socials-newsletter'
-        )[0],
-        activePremiumFeaturing: activeGameFeaturings.filter(
-          gameFeaturing => gameFeaturing.featuring === 'gdevelop-banner'
-        )[0],
-      };
     },
     [gameFeaturings]
   );
 
-  const getActiveFeaturing = React.useCallback(
+  const isMarketingPlanActive = React.useCallback(
     (marketingPlan: MarketingPlan) => {
-      switch (marketingPlan.id) {
-        case 'featuring-basic':
-          return activeBasicFeaturing;
-        case 'featuring-pro':
-          return activeProFeaturing;
-        case 'featuring-premium':
-          return activePremiumFeaturing;
-        default:
-          return null;
-      }
+      if (!activeGameFeaturings) return false;
+      const includedMarketingPlanFeaturings = marketingPlan.includedFeaturings;
+
+      // A marketing plan is considered active if it has all the inluded featurings active.
```

Comment:
Note that it might be worth getting this out of the component and pass an additional parameter, if you think this can helps making the component less busy.

Reply:
oops I missed that comment 👓  will be for a next PR if we work on this component again!